api/app/Modules/Library/Projectors/DraftLyricsProjector.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Modules\Library\Projectors;
+
+use App\Modules\Library\Events\Drafts\Lyrics\DraftLyricsChanged;
+use App\Modules\Library\Events\Drafts\Lyrics\DraftLyricsCreated;
+use App\Modules\Library\Events\Drafts\Lyrics\DraftLyricsDeleted;
+use App\Modules\Library\Events\Drafts\Lyrics\DraftLyricsPublished;
+use App\Modules\Library\Events\Tracks\TrackLyricsChanged;
+use App\Modules\Library\Models\DraftLyrics;
+use App\Modules\Lyrics\Documents\Factory as DocumentFactory;
+use Spatie\EventSourcing\EventHandlers\Projectors\Projector;
+
+class DraftLyricsProjector extends Projector
+{
+    /**
+     * @throws \JsonException
+     * @throws \Throwable
+     */
+    public function onDraftLyricsCreated(DraftLyricsCreated $event)
+    {
+        $draftLyrics = new DraftLyrics();
+        $draftLyrics->id = $event->id;
+        $draftLyrics->track_id = $event->trackId;
+        $draftLyrics->document = $event->document;
+        $draftLyrics->saveOrFail();
+    }
+
+    /**
+     * @throws \Throwable
+     */
+    public function onDraftLyricsChanged(DraftLyricsChanged $event)
+    {
+        $draftLyrics = DraftLyrics::retrieve($event->id);
+        $draftLyrics->document = $event->document;
+        $draftLyrics->saveOrFail();
+    }
+
+    public function onDraftLyricsPublished(DraftLyricsPublished $event)
+    {
+        $draftLyrics = DraftLyrics::retrieve($event->id);
+        event(new TrackLyricsChanged($draftLyrics->track_id, $event->document));
```

Comment:
I still think that a projector should not fire new events. This should directly update the track lyrics, rather than firing this event.